sc-memory/sc-memory/sc_scs_helper.cpp

```diff
@@ -169,7 +169,23 @@ class StructGenerator
         {
           SetSCsGlobalIdtf(el.GetIdtf(), result);
         }
-        
+      }
+      else
+      {
+        ScType const & newType = el.GetType();
+        ScType const & oldType = m_ctx.GetElementType(result);
+        if (oldType.CanExtendTo(newType))
+        {
+          m_ctx.SetElementSubtype(result, *newType);
+        }
+        else
+        {
+          //Type NodeConst means that element have been called by system Id, that is why need to ignore this case as error
+          if (newType != ScType::NodeConst)
+          {
+            SC_THROW_EXCEPTION(utils::ExceptionInvalidType, "Duplicate element type for " + el.GetIdtf());
+          }
+        }
       }
```

Comment:
```cpp
if (!m_ctx.SetElementSubtype(result, *newType))
{
   SC_THROW_EXCEPTION(utils::ExceptionInvalidType, "Duplicate element type for " + el.GetIdtf());
}
```
Also as i mentioned before we should provide more error information. Which was type, which type we want to setup